specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,937 @@
+    'appearance',
+    'layout'
+  );
+
+  const [personalLayout, setPersonalLayout] = usePref(
+    'statistics',
+    'appearance',
+    'layout'
+  );
+
+  const [defaultLayout, setDefaultLayout] = useCollectionPref(
+    'statistics',
+    'appearance',
+    'defaultLayout'
+  );
+
+  const layout = {
+    [statsText('shared')]: collectionLayout,
+    [statsText('personal')]: personalLayout,
+  };
+
+  const [state, setState] = React.useState<
+    | State<
+        'AddingState',
+        {
+          readonly pageIndex: number;
+          readonly categoryIndex: number;
+        }
+      >
+    | State<
+        'PageRenameState',
+        {
+          readonly pageIndex: number | undefined;
+          readonly isCollection: boolean;
+        }
+      >
+    | State<'DefaultState'>
+    | State<'EditingState'>
+    | State<
+        'DeletingCategoryState',
+        { readonly categoryContainsCustom: boolean }
+      >
+  >({ type: 'DefaultState' });
+
+  const isAddingItem = state.type === 'AddingState';
+  const isEditing =
+    state.type === 'EditingState' ||
+    isAddingItem ||
+    state.type === 'PageRenameState';
+  const isEditingCollection = React.useRef(false);
+  const canEditIndex = (isCollection: boolean): boolean =>
+    isCollection
+      ? hasPermission('/preferences/statistics', 'edit_protected') &&
+        isEditingCollection.current
+      : !isEditingCollection.current;
+
+  const [activePage, setActivePage] = React.useState<{
+    readonly isCollection: boolean;
+    readonly pageIndex: number;
+    readonly isPageUpdated: boolean;
+  }>({
+    isCollection: true,
+    pageIndex: 0,
+    isPageUpdated: false,
+  });
+  const filters = React.useMemo(
+    () => ({
+      specifyUser: userInformation.id,
+    }),
+    []
+  );
+
+  const categoriesToFetchInitially = useCategoryToFetch(
+    activePage.isCollection ? collectionLayout : personalLayout
+  );
+
+  const allCategories = React.useMemo(() => Object.keys(urlSpec), []);
+
+  const [categoriesToFetch, setCategoriesToFetch] = useTriggerState<RA<string>>(
+    categoriesToFetchInitially
+  );
+  const backEndResponse = useBackendApi(categoriesToFetch, false);
+  const defaultBackEndResponse = useBackendApi(allCategories, false);
+  const statsSpec = useStatsSpec();
+
+  const defaultStatsSpec = useStatsSpec();
+  const defaultLayoutSpec = useDefaultLayout(defaultStatsSpec);
+
+  /* Initial Load For Collection and Personal Pages*/
+  React.useEffect(() => {
+    if (collectionLayout === undefined) {
+      setCollectionLayout(defaultLayoutSpec);
+      setCategoriesToFetch(allCategories);
+    }
+    if (personalLayout === undefined) {
+      setPersonalLayout(defaultLayoutSpec);
+      setCategoriesToFetch(allCategories);
+    }
+  }, [
+    collectionLayout,
+    defaultLayoutSpec,
+    personalLayout,
+    setCollectionLayout,
+    setPersonalLayout,
+    allCategories,
+    setCategoriesToFetch,
+  ]);
+
+  /* Set Default Layout every time page is rendered*/
```

Comment:
Comments that say what you do is not helpful - I can see that from the code (unless the code is beyond horrible)
Comments explaining why you do something is what's helpful - refactor this and others you can find